### PR TITLE
internship and ng leaderboard through channel reactions

### PR DESCRIPTION
**Pull request by elimelt**
need to add this to env:

```bash
export INTERNSHIP_CHANNEL_ID=1298121534489231421
export NG_CHANNEL_ID=1298149664708890676
```

### CODE REVIEW

File: server/leaderboard/urls.py [hunk 1]
```diff
@@ -8,4 +8,7 @@
         name="leetcode-leaderboard",
     ),
     path("github/", views.GitHubLeaderboardView.as_view(), name="github-leaderboard"),
+    path("internship/", views.InternshipApplicationLeaderboardView.as_view(), name="internship-leaderboard"),
+    path("newgrad/", views.NewGradApplicationLeaderboardView.as_view(), name="newgrad-leaderboard"),
```
Comment (Advayp) [inline server/leaderboard/urls.py:12]: Can you change `newgrad` to `new-grad`? The linter complains over on the frontend lol.

File: server/leaderboard/views.py [hunk 2]
```diff
@@ -88,3 +97,160 @@ def get_queryset(self):
...
+                    f"{INTERNSHIP_CHANNEL_ID} and {NEW_GRAD_CHANNEL_ID}"
+                )
+                return Response(status=status.HTTP_304_NOT_MODIFIED)
+
+            self._handle_stats(user_id, channel_config, 'decrement')
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        except Http404:
+            logger.error(f"User not found for discord_id: {discord_id}")
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (Advayp) [inline server/leaderboard/views.py:256]: Is 404 the right error code to use here?

File: server/leaderboard/views.py [hunk 3]
```diff
@@ -88,3 +97,160 @@ def get_queryset(self):
...
+                    f"{INTERNSHIP_CHANNEL_ID} and {NEW_GRAD_CHANNEL_ID}"
+                )
+                return Response(status=status.HTTP_304_NOT_MODIFIED)
+
+            self._handle_stats(user_id, channel_config, 'increment')
+            return Response(status=status.HTTP_201_CREATED)
+
+        except Http404:
+            logger.error(f"User not found for discord_id: {discord_id}")
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (Advayp) [inline server/leaderboard/views.py:233]: See comment below about 404 error code.

**Comment by Advayp** [comment]
Correct environment to add:
```bash
export INTERNSHIP_CHANNEL_ID=1298121534489231421
export NEW_GRAD_CHANNEL_ID=1298149664708890676
```

**Comment by elimelt** [comment]
> Correct environment to add:
> 
> ```bash
> 
> export INTERNSHIP_CHANNEL_ID=1298121534489231421
> 
> export NEW_GRAD_CHANNEL_ID=1298149664708890676
> 
> ```

oh yeah my bad, should be correct in the code though

**Comment by Advayp** [comment]
> > Correct environment to add:
> > ```shell
> > export INTERNSHIP_CHANNEL_ID=1298121534489231421
> > 
> > export NEW_GRAD_CHANNEL_ID=1298149664708890676
> > ```
> 
> oh yeah my bad, should be correct in the code though

Yeah it's all good in the code

File: server/leaderboard/views.py (hunk 2 above)
Comment (elimelt) [reply server/leaderboard/views.py:256]: like semantically? or are you worried about errors being returned when people without accounts react

Comment (Advayp) [reply server/leaderboard/views.py:256]: I was worried about it just semantically, but now I think it's probably perfectly fine to return a 404 for a missing resource

**Review by Advayp — APPROVED** [review]

**Pull request merged**